sympy/matrices/matrices.py

```diff
@@ -111,6 +114,12 @@ def __len__(self):
 class MatrixShaping(MatrixRequired):
     """Provides basic matrix shaping and extracting of submatrices"""
 
+    def _eval_col_del(self, col):
+        def entry(i, j):
+            i = i - 1 if i > col else i
+            return self[i, j]
```

Comment:
This is a nitpick, but how about `return self[i-1, j] if i > col else self[i, j]` instead of possibly setting setting `i` equal to itself when `i <= col` is True?